catalogs/catalog-common/src/main/java/com/datastrato/catalog/common/ClassProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.catalog.common;
+
+import com.datastrato.gravitino.catalog.BasePropertiesMetadata;
+import com.datastrato.gravitino.catalog.PropertiesMetadata;
+import com.datastrato.gravitino.catalog.PropertyEntry;
+import com.datastrato.gravitino.catalog.hive.HiveCatalogPropertiesMeta;
+import com.datastrato.gravitino.catalog.hive.HiveSchemaPropertiesMetadata;
+import com.datastrato.gravitino.catalog.hive.HiveTablePropertiesMetadata;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergCatalogPropertiesMetadata;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergSchemaPropertiesMetadata;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergTablePropertiesMetadata;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * {@link ClassProvider} holds the set of classes that are required for the query engine to use
+ * Gravitino, currently, it mainly focuses on the classes that are required for the property
+ * metadata.
+ */
```

Comment:
Can you add more comments about the reasons why we need to build this package.

Reply:
Added.